tests/test_speed.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
```

Comment:
Add tests for consistency of wavefunction/expectation value with different `apply_xxx` methods.

Reply:
right now, the method is about the same/ sometimes slower than apply_batch_gate which might be due to a lack of optimization of vmap for complex tensors, however we should keep the method for future purposes